docs/services.md

```diff
@@ -26,6 +26,103 @@ systemctl cat bitcoind
 systemctl show bitcoind
 ```
 
+# Setup clightning database replication
+
+Clightning can write an additional, replica database to a local directory. The
+local directory can also be used to write to custom targets mounted at that
+path, like external HDD's, or NFS/SMB shares. Nix-bitcoin has built-in support
+for replicating to a remote filesystem over SSH (using SSHFS). Backups can also
+be automatically encrypted using gocryptfs.
+
+Note: You should also backup `hsm_secret` (located at
+`/var/lib/clightning/bitcoin/hsm_secret` in most cases) separately, manually
+and/or using the `services.backups` module.
+
+## SSHFS
+
+1. Append to your `configuration.nix`
+
+    ```
+    services.clightning.replication = {
+      enable = true;
+      sshfs.destination = "user@hostname:directory";
+      encrypt = true;
+    };
+    programs.ssh.knownHosts."hostname".publicKey = "<ssh public key from running `ssh-keyscan` on the host>";
+    ```
+
+    Leave out the `encrypt` line if you want to store data on your destination
+    in plaintext and adjust `user`, `hostname` and `directory` as necessary.
+
+2. Deploy new `configuration.nix`
+
+3. Authorize the public key on your destination by copying the contents of
+   `$secretsDir/clightning-replication-ssh.pub` to the `authorized_keys` file of
+   `user` or by using `ssh-copy-id`.
+
+4. You can restrict the nix-bitcoin node's capabilities on the SSHFS target
+   using OpenSSH's builtin features, as detailed
+   [here](https://serverfault.com/questions/354615/allow-sftp-but-disallow-ssh).
+
+   To easily get this working on NixOS, you should add the following to the
+   SSHFS target's configuration:
+
+    ```
+    systemd.tmpfiles.rules = [
+      "d '/var/backup/<user>' 0755 root root - -"
+      "d '/var/backup/<user>/writeable' 0700 <user> sftponly - -"
+    ];
+
+    services.openssh = {
+      extraConfig = ''
+        Match group sftponly
+          ChrootDirectory /var/backup/%u
+          X11Forwarding no
+          AllowTcpForwarding no
+          AllowAgentForwarding no
+          ForceCommand internal-sftp
+      '';
+    };
+
+    users.groups.sftponly = {};
+    users.users.<user> = {
+      shell = "${pkgs.coreutils}/bin/false";
+      extraGroups = [ "sftponly" ];
+    };
+    ```
+
+    If the `<user>` is set to `nb-replication` and the targets hostname is
+    `hostname`, then the corresponding `sshfs.destination` on the nix-bitcoin
+    node is `"nb-replication@hostname:/var/backup/nb-replication/writeable"`.
```

Comment:
I'm recklessly running e086fc567baf394eb5d5bf873eec9f614f36c6b2 in production now and one thing I noticed is that due to `ChrootDirectory /var/backup/%u` this path should be `/writable` instead of `/var/backup/nb-replication/writeable`.